### PR TITLE
refactor(poller): BREAKING CHANGE - remove serializer package, remove GetSplitData from Fetcher/poller

### PR DESCRIPTION
**Pull request by jh2oman**
Followup PR to discussions https://github.com/godaddy/split-go-serializer/pull/18#discussion_r387999866 and https://github.com/godaddy/split-go-serializer/pull/18#discussion_r387993847

### CODE REVIEW

File: poller/poller.go [hunk 1]
```diff
@@ -107,11 +106,6 @@ func (poller *Poller) pollForChanges() {
 
 }
 
-// GetSplitData returns split data cache results
```
Comment (rohanmurthy) [inline poller/poller.go]: Why do we need to get rid of this? I was planning on using it for the change I am working on.

Comment (jh2oman) [reply poller/poller.go]: Its unused and not a documented part of the package. I could just make it an internal function?

Comment (rohanmurthy) [reply poller/poller.go]: Sure, that works!

Comment (jh2oman) [reply poller/poller.go]: Actually, can you add it back in as internal as part of your PR? I think that makes more sense

File: README.md [hunk 2]
```diff
@@ -23,20 +23,17 @@ A Go module which fetches split definitions and segments from Split.io and seria
 
 Use this Go module in your server-side Go environment. The serializer exposes:
 1. a `Poller` that periodically requests raw experiment configuration data from Split.io. Requests happen in the background and the poller caches the latest data in local memory.
-1. a `Serializer` that reads from the poller's cache, serializes the data, and returns it in a script to be injected into a client's HTML.
 
 ### Instantiation
 
-Create an instance of `Poller` and `Serializer` by importing the `poller` and `serializer` package of this module and calling the `NewPoller` and `NewSerializer` function with some parameters :
+Create an instance of `Poller`  by importing the `poller` package of this module and calling the `NewPoller` function with some parameters :
```
Comment (rohanmurthy) [inline README.md]:
```suggestion
Create an instance of `Poller` by importing the `poller` package of this module and calling the `NewPoller` function with some parameters :
```

File: poller/poller_test.go [hunk 3]
```diff
@@ -48,6 +49,11 @@ func (splitio *mockSplitio) GetSegmentsForSplits(splits map[string]dtos.SplitDTO
 	return nil, 0, fmt.Errorf("Error from splitio API when getting segments")
 }
 
+// getSplitData helper returns split data cache results
+func getSplitData(poller *Poller) SplitData {
```
Comment (wendychiang) [inline poller/poller_test.go]: So @rohanmurthy will refactor this right?

Comment (jh2oman) [reply poller/poller_test.go]: yep! he'll use the new private function instead

**Pull request merged**